napalm_junos/utils/junos_views.yml

```diff
@@ -277,7 +277,7 @@ junos_bgp_config_peers_view:
     import_policy: {import: unicode}
     export_policy: {export: unicode}
     local_address: {local-address: unicode}
-    local_as: {ocal-as: int}
+    local_as: {local-as/as-number: int}
```

Comment:
What Junos version do you have @ckishimo? On MX series, running 15.x, I can't see this hierarchy, but the local AS number directly under the `local-as` tag, so this should be just `local_as: {local-as: int}`. Do you have any device where the XML is something like:

```xml
<local-as>
  <as-number>1234</as-number>
</local-as>
```

Reply:
I was using Junos: 14.1R1.10 for this kind of tests...
```
                        <neighbor>
                            <name>20.20.20.20</name>
                            <description>routerB</description>
                            <local-address>22.22.22.22</local-address>
                            <peer-as>12345</peer-as>
                            <local-as>
                                <as-number>12345</as-number>
                            </local-as>
                        </neighbor>
```

Reply:
Interesting. Hail for consistency!
Can you please check if `local_as: {local-as | local-as/as-number: int}` works fine? That would be the easy way... fingers crossed - otherwise it might be harder :|

Reply:
Nop... with `local_as: { local-as | local-as/as-number: int}`, you get this:
```
Traceback (most recent call last):
  File "client_junos_bgp.py", line 21, in <module>
    res = dev.get_bgp_config()
  File "/home/ckishimo/python/devel/git/devel-config/devel-config/napalm_junos/junos.py", line 904, in get_bgp_config
    bgp_items = bgp.items()
  File "/home/ckishimo/python/devel/git/devel-config/env/local/lib/python2.7/site-packages/jnpr/junos/factory/table.py", line 174, in items
    return list(zip(self.keys(), self.values()))
  File "/home/ckishimo/python/devel/git/devel-config/env/local/lib/python2.7/site-packages/jnpr/junos/factory/table.py", line 166, in values
    return [list(this.items()) for this in self]
  File "/home/ckishimo/python/devel/git/devel-config/env/local/lib/python2.7/site-packages/jnpr/junos/factory/view.py", line 121, in items
    return zip(self.keys(), self.values())
  File "/home/ckishimo/python/devel/git/devel-config/env/local/lib/python2.7/site-packages/jnpr/junos/factory/view.py", line 117, in values
    return [getattr(self, field) for field in self.keys()]
  File "/home/ckishimo/python/devel/git/devel-config/env/local/lib/python2.7/site-packages/jnpr/junos/factory/view.py", line 268, in __getattr__
    raise RuntimeError("Unable to handle field:'%s'" % name)
RuntimeError: Unable to handle field:'local_as'
```
...and I started it as a typo :smile:

Reply:
It works without the type like:
`local_as: local-as | local-as/as-number`
@mirceaulinic  Can you check in your MX running 15.x ?

Reply:
Yeah, just do the same as under the config table, `local_as: local-as | local-as/as-number` and then convert the value to integer in the body of the method.

Reply:
@mirceaulinic: I can see the local AS number directly under the `local-as` tag when using any `show bgp xx | display xml`. However I can see the tag `local-as/as-number` if you `get: protocols/bgp/group` when getting the value from configuration. Is it different in Junos 15.x ?
If we leave it as `local_as: {local-as/as-number: int}` what is the error you get ?

Reply:
This is not a matter of what I see, this is more of the tests must pass with the existing data, which is collected from older Junos versions. In other words, `local_as: {local-as/as-number: int}` is not acceptable, and we need to support both, the solution being `local_as: local-as | local-as/as-number` with the value cast to `int`, inside the getters.